File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.go
func minDeletionSize(strs []string) int {
    if len(strs) == 1 {
        return 0
    }

    countDelete := 0
    for j := 0; j < len(strs[0]); j++ {
        for i := 1; i < len(strs); i++ {
            if strs[i][j] < strs[i-1][j] {
                countDelete++
                break
            }
        }
    }

    return countDelete
}

// array, string
// time: O(nm)
// space: O(1)

File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.py
class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        if len(strs) == 0:
            return 0
        
        countDelete = 0
        for j in range(len(strs[0])):
            for i in range(1, len(strs)):
                if strs[i][j] < strs[i-1][j]:
                    countDelete += 1
                    break
        
        return countDelete

        